### DISCUSSION TITLE
Validating causal discovery

### DISCUSSION THREAD

**Discussion by jjakenichol** [jakobrunge/tigramite#165] [Q&A] [answered]
Hi, this is a question about causal discovery generally, but an answer specific to PCMCI will work for me. Absent a ground truth, how does one validate the results of a causal discovery algorithm? It's clear that ground truth is the only way to be certain, but it seems that the purpose of causal discovery is to infer causal structures that we don't already understand. I get this question a lot from my machine learning peers who are accustomed to having ROC curves and so on. 

The best short answer I've come up with is this, and ML people seem to find it dissatisfying: causal identifiablity has been proven when we make certain assumptions about the data and the underlying causal structure (referring to the proofs like Theorem 4.5, the identifiability of nonlinear additive noise models in Peters et al's Elements of Causal Inference or Pearl 1988 Chapter 3). The validation of causal discovery is then more a process of validating those assumptions than it is validating the method's results. Of course, expert domain knowledge is critical for doing this in whatever the relevant field is. In climate, we'd want to understand the physics in the system and the functional forms of the dependencies, as well as make sure that the results are not violating any physical laws or other governing dynamics we understand well.

How does that sound? Am I missing something critical? Are there analytical ways to validate these?

One idea I've thought about is taking a mean over the AICs computed for each link in the parcorr case. That might give an average likelihood of how well the graph "fits" the data, but it still doesn't validate "causality."

Thanks for reading and I appreciate any thoughts you all have.

**Comment by jakobrunge** [comment, marked as answer]
Theoretical results are good for the asymptotic regime and then it is indeed about validating assumptions. For short sample sizes, it is about benchmarking, that's what the CauseMe.net platform is about. To some extend this topic is also discussed in this paper here: https://www.nature.com/articles/s41467-019-10105-3
